unit-tests/src/test/scala/tests/SuiteSuite.scala

```diff
@@ -66,10 +66,12 @@ object SuiteSuite extends Suite {
         }
 
       case exc: Throwable =>
-        if (exc == AssertionFailed) // got case object, expected case class
-          throw AssertionFailed("expected message yet none found")
-        else
-          throw exc
+        exc match {
+          case _: AssertionFailed =>
+            // got case object, expected case class
```

Comment:
The comment is in contradiction with the code. Testing for `case _: AssertionFailed` means that we got an instance of the class, not of the case object.